ngraph/core/shape_inference/include/shuffle_channels_shape_inference.hpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <openvino/op/shuffle_channels.hpp>
+
+namespace ov {
+namespace op {
+namespace v0 {
+
+template <class T>
+void shape_infer(const ShuffleChannels* op, const std::vector<T>& input_shapes, std::vector<T>& output_shapes) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 1 && output_shapes.size() == 1);
+    const auto& input_shape = input_shapes[0];
+    NODE_VALIDATION_CHECK(op, op->m_group >= 1, "The 'group' parameter must be greater or equal to 1.");
+    if (input_shape.is_static()) {
```

Comment:
We only need input shape to have static rank and shape[axis] to be static to perform the checks